src/Cardano/Wallet/Primitive.hs

```diff
@@ -283,6 +294,84 @@ restrictedTo (UTxO utxo) outs =
     UTxO $ Map.filter (`Set.member` outs) utxo
 
 
+-- * Slotting
+
+-- | Hard-coded for the time being
+slotsPerEpoch :: Natural
+slotsPerEpoch = 21600
+
+-- | A slot identifier is the combination of an epoch and slot.
+data SlotId = SlotId
+  { epochIndex :: !Word64
+  , slotNumber :: !Word16
+  } deriving stock (Show, Eq, Ord, Generic)
+
+instance NFData SlotId
+
+instance Enum SlotId where
+    toEnum i
+        | i < 0 = error "SlotId.toEnum: bad argument"
+        | otherwise = slotIncr (fromIntegral i) (SlotId 0 0)
+    fromEnum (SlotId e s)
+        | n > fromIntegral (maxBound @Int) =
+            error "SlotId.fromEnum: arithmetic overflow"
+        | otherwise = fromIntegral n
+      where
+        n :: Natural
+        n = fromIntegral e * fromIntegral slotsPerEpoch + fromIntegral s
+
+-- | Add a number of slots to an (Epoch, LocalSlotIndex) pair, where the number
+-- of slots can be greater than one epoch.
+slotIncr :: Natural -> SlotId -> SlotId
```

Comment:
Why `Natural`, rather than `Word64`?

Reply:
I kept the original type-signature defined earlier. I believe that the rational was that `Natural` are unbounded, though in practice, a `Word64` is more than fine indeed.